crates/revm/src/context/inner_evm_context.rs

```diff
@@ -273,15 +246,10 @@ impl<DB: Database> InnerEvmContext<DB> {
                 created_address,
             } => (input.clone(), initcode.clone(), *created_address),
             EOFCreateKind::Tx { initdata } => {
-                // Use nonce from tx (if set) or from account (if not).
+                // Use nonce from tx.
                 // Nonce for call is bumped in deduct_caller
                 // TODO(make this part of nonce increment code)
-                let nonce = self.env.tx.nonce.unwrap_or_else(|| {
-                    let caller = self.env.tx.caller;
-                    self.load_account(caller)
-                        .map(|(a, _)| a.info.nonce)
-                        .unwrap_or_default()
-                });
```

Comment:
I moved the configuration of disabled nonce checks to the `CfgEnv` such that the transaction itself always requires a `nonce` value to be provided. If someone wants to disable nonce checks, they can do so for the whole `Evm`, similar to how it works for disabling other checks.

This voids the need for the `trait Transaction` to return an `Option<u64>` and I believe is in line with the overall design philosophy for having compile-time disable for checks like these.